src/rootReducer.js

```diff
@@ -27,25 +28,35 @@ const buildBoard = (state, numberOfElements) => {
 const markMatchedCards = (state, matchValue) => {
   const newGameBoard = state.gameBoard.map(card => {
     if(card.value === matchValue) {
-      card.visible = false
       card.matched = true
       return card
     } else {
       return card
     }
   })
-  return { ...state, gameBoard: newGameBoard }
+  if(validateGameEnd(newGameBoard)) {
+    return { ...state, phase: GAME_END, gameBoard: newGameBoard }
+  } else {
+    return { ...state, gameBoard: newGameBoard }
+  }
+}
+
+const validateGameEnd = (board) => {
+  return board.filter(card => !card.matched).length === 0 ? true : false
```

Comment:
`board.filter(card => !card.matched).length === 0` samo to już zwróci true lub false nie potrzebujesz dodatkowego if-a

Reply:
ok, poprawiam